packages/turbo/ios/RNVisitableViewController.swift

```diff
@@ -24,36 +24,82 @@ public protocol RNVisitableViewControllerDelegate {
   
 }
 
-class RNVisitableViewController: VisitableViewController {
-  
-  public var delegate: RNVisitableViewControllerDelegate?
-  
+class RNVisitableViewController: UIViewController, Visitable {
+  public var visitableViewControllerDelegate: RNVisitableViewControllerDelegate?
+
+  open weak var visitableDelegate: VisitableDelegate?
+  open var visitableURL: URL!
+
+  private var reactViewController: UIViewController? = nil
+
+  public convenience init(url: URL) {
+    self.init()
+    self.visitableURL = url
+  }
+    
+  public convenience init(reactViewController: UIViewController?, visitableViewControllerDelegate: RNVisitableViewControllerDelegate?) {
+    self.init()
+    self.reactViewController = reactViewController
+    self.visitableViewControllerDelegate = visitableViewControllerDelegate
+  }
+
+  // MARK: View Lifecycle
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    view.backgroundColor = UIColor.white
+    installVisitableView()
+  }
+
   override func viewWillAppear(_ animated: Bool) {
     super.viewWillAppear(animated)
-    delegate?.visitableWillAppear(visitable: self)
+    visitableViewControllerDelegate?.visitableWillAppear(visitable: self)
```

Comment:
Hm. I think we'll be missing some delegations here, right?

Previously we had _2_ delegates being called here; `delegate` and then we called `super` as well, which pointed to the `VisitableViewController`.

In `VisitableViewController` there was:

```
super.viewDidAppear(animated)
visitableDelegate?.visitableViewDidAppear(self)
```

So previously we called two delegates:
- `VisitableViewController.visitableDelegate`
- `delegate`

Now we call one:
- `visitableViewControllerDelegate`

Reply:
I'm kinda thinking if the code here works at all - have you tested it? As I read it, no lifecycle methods would be called on `Session` which essentially breaks everything, right?

Reply:
It is working because of this line of code in the RNVisitableView:
```
  @objc var url: NSString = "" {
    didSet {
      if(url != oldValue) {
        visit()
      }
    }
  }
```

It is called on every page visit, so if we call `visitableDelegate?.visitableViewWillAppear(self)` we're gonna call `visit()` twice which we want to avoid. I think we want to revert the behavior of didSet which was introduced in #107  and let's delegate that to the visitableDelegate.

Reply:
Okay, without that we won't initialize the visit at all, `visitableViewWillAppear` handles cancelling back gestures etc so actually we shouldn't call this twice.